### ISSUE TITLE
Forbid meaningless `raise` in `except`

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#366]
# Rule request

## Thesis

Whenever we see code like:

```python
try:
    sm = 1 / 0
except ZeroDivisionError:
    raise
```

or 

```python
try:
    sm = 1 / 0
except ZeroDivisionError as ex:
    raise ex
```

We should raise a violation.
It only applies to `except` with `raise` as first and only element. Things we should allow:

```python
try:
    sm = 1 / 0
except ZeroDivisionError:
    sentry.report()
    raise
```

And

```python
try:
    sm = 1 / 0
except ZeroDivisionError:
    raise ValueError()
```

## Reasoning

There's no need to do so. This will just reraise the same exception again. All `except` node can be removed with no effect.

**Comment by sobolevn** [comment]
Should be allowed too:


```python
try:
    ...
except Exception as ex:
     raise TypeError from ex
```

**Comment by kxepal** [comment]
Well, I have a case for you:
```
try:
  ...
except asyncio.CancelledError:
  raise
except:
  sentry.capture_exception()
  raise
```
The case here is that we don't want to capture `asyncioCancelledError`'s at all since those actually means two things:
- client was disconnected
- server is on the stop way and shutdown all the tasks
Both are not relevant for us because these things happens and they are fine.  And certaily, we don't want to receive notifications about that some user hit the Stop button on their browser in the middle of page load.

**Comment by sobolevn** [comment]
This might be a case for `noqa`. If really think that this is a best option.

**Comment by kxepal** [comment]
Sounds like a hack /:

**Comment by sobolevn** [comment]
The thing is, I see several rule violations here: and I am not sure why do you have this code in the first place.

What you do here is an implicit `isinstance` condition and you code can be refactored like follows:

```python
try:
    ...
except Exception as exc:
     if not isinstance(exc, asyncio.CancelledError):
         sentry.capture_exception()
     raise
```

But, for some reason you also capture `GeneratorStop` / `KeyboardInterupt` and other non-exceptions here. If you do that on purpose you can change your code to be: `except BaseException as exc:`
